src/main/kotlin/com/jetbrains/snakecharm/SnakemakeRuleInheritanceMarkerProvider.kt

```diff
@@ -52,17 +54,14 @@ class SnakemakeRuleInheritanceMarkerProvider : RelatedItemLineMarkerProvider() {
         result: MutableCollection<in RelatedItemLineMarkerInfo<*>>
     ) {
         val currentFile = element.containingFile
-        val identifier = (if (element is SmkUse) element.getIdentifierLeaf() else element.nameIdentifier) ?: return
-        val names = if (element is SmkUse) element.getProducedRulesNames().map { it.first } else listOf(
-            element.name ?: return
-        )
-        val e = StubIndex.getInstance().getAllKeys(KEY, element.project)
+        val identifier = (if (element is SmkUse && element.nameIdentifierIsWildcard()) element.nameIdentifier?.firstChild else element.nameIdentifier) ?: return
```

Comment:
Copy-paste of some complicated logic like in line 34, please use some API instead.